File: oddbins/page_get_over_proxy.py
#!/usr/bin/env python

import requests

proxies = {
    'http':'http://127.0.0.1:8888',
    'https':'http://127.0.0.1:8888'
}

r = requests.get('http://google.com', proxies=proxies)

if r.status_code == requests.codes.ok:
    print("Successfully downloaded google index page!")
elif r.status_code == 407:
    print("Failed to reach google, hitting some form of authenticated proxy.")
    print("----------")
    print("r.text")
    print("----------")
else:
    print("Failed to reach google, non-200 HTTP status code is {}".format(r.status_code))

File: oddbins/page_get_over_proxy.rs
extern crate reqwest; // add reqwest = "0.8.4" to your [dependencies] section of your Cargo.toml

fn main() {
    let client = reqwest::Client::builder()
        .proxy(reqwest::Proxy::http("http://127.0.0.1:8888").unwrap())
        .build()
        .unwrap()
        .get("http://google.com")
        .send()
        .unwrap();

    if client.status().is_success() {
        println!("Successfully downloaded the google index!");
    } else {
        println!("Failed to reach google, got the following status code: {}.", client.status());
    }

    // without a proxy
    let res = reqwest::get("http://google.com").expect("Failed to reach google.");

}

